🎨 [New Terms]: Java Math Methods: `logarithmic functions` (#902) (#1181)
Codecademy/docs: content/java/concepts/math-methods/terms/log1p/log1p.md

---

> Title: '.log1p()'

Reviewer: ```suggestion
'Math.log1p()'
```

Description: 'Returns the natural logarithm (base e) of the sum of a double value as a parameter and 1.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Arithmetic'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`.log1p()`** method returns `ln (a + 1)`, the natural logarithm of a + 1.

Reviewer: ```suggestion
The **`Math.log1p()`** method returns the natural logarithm of a number plus 1.
```

## Syntax

```pseudo
Math.log1p(a);
```

> - The `.log1p()` method takes one parameter `a`.

Reviewer: This line is not necessary as this function has been explained previously.

> - If `a` is `NaN`, then return value is `NaN`.

Reviewer: Lines 23-27 should be the same as the ones in the entry for the previous two functions.

- If `a` is `negative number`, then return value is `NaN`.
- If `a` is `positive number`, then return value is `double`.
- If `a` is `positive / negative zero`, then return value is `positive infinity`.
- If `a` is `positive infinity`, then return value is `positive infinity`.

## Example

> The following example demonstrates using `.log1p()` to find natural logs:

Reviewer: ```suggestion
The following example demonstrates using `Math.log1p()` to find natural logs:
```

```java
import java.lang.Math;

public class Main {
     public static void main(String args[]) {
        double a = 23;
        double b = -42;
        double c = 0;

        System.out.println("The result of log1p(" + a + ") is: " + Math.log1p(a));
        System.out.println("The result of log1p(" + b + ") is: " + Math.log1p(b));
        System.out.println("The result of log1p(" + c + ") is: " + Math.log1p(c));
    }
}
```

This will produce the following output:

```shell
The result of log1p(10) is: 3.1780538303479458
The result of log1p(-6) is: NaN
The result of log1p(0) is: -Infinity
```

## Codebyte Example

> The following example is runnable and return `ln (a + 1)`:

Reviewer 2: ```suggestion
The following example is runnable and returns `ln (a + 1)`:
```
